File: Java/String-2/catDog.java
/*
 * Return true if the string "cat" and "dog" appear the same
 * number of times in the given string.
 */

public boolean catDog(String str) {
  int catcount = 0;
  int dogcount = 0;
  
  // Find out how many times cat and dog appear in the string
  for(int i = 0; i < str.length() - 2; i++){
    if(str.substring(i, i + 3).equals("cat"))
      catcount++;
    if(str.substring(i, i + 3).equals("dog"))
      dogcount++;
  }
  return (catcount == dogcount);
}

File: Python/String-2/cat_dog.py
# Return True if the string "cat" and "dog" appear the same number of times
# in the given string.

def cat_dog(str):
  cat_count = 0
  dog_count = 0
  
  # Check to see how many times cat and dog appear in the string
  for i in range(len(str) - 2):
    if (str[i:i + 3] == 'cat'):
      cat_count += 1
    if (str[i:i + 3] == 'dog'):
      dog_count += 1
  
  return (cat_count == dog_count)
